mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSArea.kt

```diff
@@ -49,23 +51,29 @@ data class MPSArea(val repository: SRepository) : IArea {
     }
 
     override fun getReference(): IAreaReference {
-        TODO("Not yet implemented")
+        return this
     }
 
     override fun resolveArea(ref: IAreaReference): IArea? {
-        TODO("Not yet implemented")
+        return takeIf { ref == it }
     }
 
     override fun <T> executeRead(f: () -> T): T {
         var result: T? = null
         repository.modelAccess.runReadAction {
             result = f()
         }
-        return result as T
+        return result!!
```

Comment:
Can't result be immediately returned when calling `f()`? That would avoid the bangs.

Reply:
return is not allowed inside of runReadAction according to the kotlin compiler. Only `return@runReadAction`, but this would not help in this case.